File: cpp_files/raylib/shape_survivors/src/drops/drop.cpp
#include "../systems.hpp"
#include "../utils.hpp"
#include "drop.hpp"

static const float DROP_RADIUS = 20.0f;

Drop::Drop(const DropType drop_type, const Vector2& start_pos)
    : state({.active = true}), position({.x = start_pos.x, .y = start_pos.y}),
      circle({.radius = DROP_RADIUS}), color({.tint = ORANGE}),
      text({.text = "", .font_size = 20.0f, .spacing = 2.0f, .tint = WHITE}) {
    //
    amount = 0;
    type   = drop_type;

    switch (type) {
        case DropType::XP:
            text.text = "XP";
            amount    = 20;
            break;

        case DropType::HP:
            text.text = "HP";
            amount    = 10;
            break;

        case DropType::NONE:
            state.active = false;
            break;
    }
}

void Drop::draw(void) const {
    Systems::draw_circle(position, circle, color);
    Utils::center_and_draw_text(Utils::OriginType::CENTER,
                                {position.x, position.y, circle.radius * 2, circle.radius * 2},
                                text);
}

File: python_files/raylib_files/shape_survivors/drop.py
from enum import Enum, auto

import pyray as p

import components
import systems


class DropType(Enum):
    NONE = auto()
    XP = auto()
    HP = auto()


DROP_RADIUS: float = 20.0


class Drop:
    def __init__(self, drop_type: DropType, start_pos: p.Vector2) -> None:
        self.state: components.State = components.State(active=True)

        self.position: components.Position = components.Position(
            start_pos.x, start_pos.y
        )

        self.circle: components.Circle = components.Circle(radius=DROP_RADIUS)

        self.color: components.Color = components.Color(tint=p.ORANGE)

        self.text: components.Text = components.Text(
            text="", font_size=20.0, spacing=2.0, tint=p.WHITE
        )

        self.amount: int = 0
        self.type: DropType = drop_type

        if self.type == DropType.XP:
            self.text.text = "XP"
            self.amount = 20
        elif self.type == DropType.HP:
            self.text.text = "HP"
            self.amount = 10
        elif self.type == DropType.NONE:
            self.state.active = False
        else:
            raise ValueError("unknown drop type")

    def draw(self) -> None:
        systems.draw_circle(self.position, self.circle, self.color)

        systems.draw_centered_text(
            self.position, self.circle.radius * 2, self.circle.radius * 2, self.text
        )
